### PR TITLE
Adds Quick Start Docs for Different Platforms and SmolLM2-135M YAML

### PR DESCRIPTION
**Pull request by vishalbakshi**
This PR adds a `docs` folder with a `Quick Starts.md` file which contains instructions to run the original llm-foundry Quick Start (but using our new installation instructions) curated for different platforms (Modal, Colab Pro, Paperspace Gradient Notebook and Mac (TBD)). 

This PR also adds a `scripts/train/yamls/pretrain/smollm2-135m.yaml` file to allow for SmolLM2-135m training. The details of this file will likely change as we make our training scripts more battle-tested.

Request for reviewers: please attempt the instructions in `Quick Starts.md` and suggest changes.

### CODE REVIEW

File: docs/Quick Starts.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+```
+../../notebooks/train_smollm2-135m.sh
+```
+You should see the output logs as described in the Modal section.
+
+### Mac
+[top](#quick-starts)
+
+TBD
```
Comment (gsganden) [inline docs/Quick Starts.md]: I've confirmed that this command works on my M1:

```
composer train/train.py \
  train/yamls/pretrain/mpt-125m.yaml \
  variables.data_local=my-copy-c4 \
  train_loader.dataset.split=train_small \
  eval_loader.dataset.split=val_small \
  max_duration=10ba \
  eval_interval=0 \
  save_folder=mpt-125m \
  model.attn_config.attn_impl=torch \
  model.loss_fn=torch_crossentropy \
  precision=FP32 \
  train_loader.num_workers=0 \
  eval_loader.num_workers=0 \
  global_train_batch_size=32 \
  device_train_microbatch_size=8 \
  train_loader.prefetch_factor=null \
  eval_loader.prefetch_factor=null \
  train_loader.persistent_workers=false \
  eval_loader.persistent_workers=false
```

I suspect it could be trimmed down.

I haven't been able to get `composer eval/eval.py` working. It hits `torch.cuda.current_device()` in `composer/distributed/dist_strategy.py` and then errors because torch was compiled without CUDA. Cursor didn't solve that problem immediately, and I'm thinking that moving on to other tasks would be a better use of my time than continuing to debug.

Comment (vishalbakshi) [reply docs/Quick Starts.md]: Sounds good thanks @gsganden---I'll add this to the .md file and we can call it good for now for Mac.

**Review by galopyz — APPROVED** [review]
I was able to run it on 3090 following using the script without any issue.

**Comment by matdmiller** [comment]
I was able to run in modal and locally after converting the modal image definition to a Dockerfile. I wouldn't hold up merging this PR, but I do think at some point it would be good to have a good cleaned up dockerfile in this repo that suits us and then use that in modal instead of their proprietary image syntax. Below is a cleaned up and trimmed down version of my dockerfile for future reference.

```
FROM mambaorg/micromamba:latest

USER root

# Install git and other dependencies
RUN apt-get update && apt-get install -y git nano

# Clone llm-foundry repo and set up environment
RUN git clone https://github.com/LocalResearchGroup/llm-foundry.git /llm-foundry && \
    cd /llm-foundry && \
    micromamba create -n llm-foundry python=3.12 uv cuda -c nvidia/label/12.4.1 -c conda-forge && \
    export UV_PROJECT_ENVIRONMENT=/opt/conda/envs/llm-foundry && \
    micromamba run -n llm-foundry uv python pin 3.12 && \
    micromamba run -n llm-foundry uv sync --extra dev --extra gpu && \
    micromamba run -n llm-foundry uv sync --extra flash && \
    micromamba run -n llm-foundry uv pip install -e .[gpu] #not sure if [gpu] part is required or not

ENV UV_PROJECT_ENVIRONMENT=/opt/conda/envs/llm-foundry
ENV CONDA_DEFAULT_ENV=llm-foundry
ENV PATH=/opt/conda/envs/llm-foundry/bin:$PATH

WORKDIR /llm-foundry

# Initialize conda in bash and activate environment by default
RUN echo "eval \"\$(micromamba shell hook --shell bash)\"" >> ~/.bashrc && \
    echo "micromamba activate llm-foundry" >> ~/.bashrc

# Default shell with environment activated
CMD ["/bin/bash"]
```

**Comment by vishalbakshi** [comment]
Thanks @matdmiller I have tested out the quick start with your Dockerfile on Modal (after removing `[gpu]` after `pip install -e.`) and it works. I have updated the Dockerfile in this branch and the Quick Starts.md accordingly. I say we merge this into main as it gives folks a good place to start with, and then make future updates to the Dockerfile as needed in a separate branch/PR.

**Comment by matdmiller** [comment]
Cool, sounds good to me. I think we should merge this now. It seems good from my testing. We can always add updates later as needed. We're not modifying critical/core code that could break stuff anyways. I don't think others are using the existing dockerfile and if they are the dependencies are possibly wrong/don't match our lock file so shouldn't be used anyways.

I also have some slightly annoying code in place right now in my local copy because this isn't merged yet that I would like to remove :) that points at this PR instead of main.

**Review by matdmiller — APPROVED** [review]

**Comment by vishalbakshi** [comment]
@matdmiller done!
